apps/web/src/middlewares/ab-test-middleware.ts

```diff
@@ -0,0 +1,53 @@
+import { EXPERIMENTAL_FEATURES, EXPERIMENTAL_FEATURE_FLAG_MAP, FeatureRollOutConfig } from 'config/experminetalFeatures'
+import { NextFetchEvent, NextResponse } from 'next/server'
+import { EnumValues, MiddlewareFactory, ModifiedNextReq } from './types'
+
+type FeatureKeys = EnumValues<typeof EXPERIMENTAL_FEATURES>[]
+
+const ctxKey = (key: EXPERIMENTAL_FEATURES) => `ctx-${key.toLowerCase()}`
```

Comment:
```suggestion
const getFeatureCookieKey = (key: EXPERIMENTAL_FEATURES) => `experimental-${key.toLowerCase()}`
```

Also move this function into `config/experminetalFeatures`.  Reason: always remember single source of truth. When changing the key, never change it in more than one places